apps/src/code-studio/initApp/project.js

```diff
@@ -586,22 +586,40 @@ var projects = (module.exports = {
       this.updateChannels_(callback);
     }
   },
-  setLibraryDetails(newName, newDescription, publishing = undefined) {
+  /**
+   * Updates the current channel's library details.
+   *
+   * @param {Object} config - Object containing library details.
+   * @param {string} config.newName
+   * @param {string} config.newDescription
+   * @param {string} config.newVersionId - S3 version ID for the current library version. Passing this value as -1 will nullify libraryLatestVersion.
+   * @param {boolean} config.publishing - true if library is being published, false if library is being unpublished, undefined otherwise.
+   */
+  setLibraryDetails(config = {}) {
     current = current || {};
+    const {newName, newDescription, newVersionId, publishing} = config;
```

Comment:
It may make more sense to allow config to continue to use `{libraryName, libraryDescription, latestLibraryVersion}` - that way when setting the library details, we can use the spread operator.
i.e.
`setLibraryDetails({...originalDetails, ...{libraryName: "myCoolLibrary"}})`

Reply:
good call -- agreed!